trestle/common/list_utils.py

```diff
@@ -110,3 +131,61 @@ def delete_list_from_list(item_list: List[TG], indices: List[int]) -> None:
 def merge_dicts(dest: Optional[Dict[str, str]], src: Optional[Dict[str, str]]) -> Dict[str, str]:
     """Merge the two dicts with priority to src."""
     return {**as_dict(dest), **as_dict(src)}
+
+
+def deep_set(dic: Dict[str, Any], path: List[str], value: Any, pop_if_none: bool = True) -> None:
```

Comment:
From the name I thought it was constructing some "set", shall we call it something more explicit like set_value_by_path?